src/CoreFoundation/CFMutableString.cs

```diff
@@ -0,0 +1,63 @@
+#if !COREBUILD
+
+using System;
+using System.Runtime.InteropServices;
+
+using ObjCRuntime;
+
+namespace CoreFoundation {
+	
+	public class CFMutableString : CFString {
+
+		[DllImport (Constants.CoreFoundationLibrary, CharSet=CharSet.Unicode)]
+		static extern unsafe /* CFMutableStringRef* */ IntPtr CFStringCreateMutableWithExternalCharactersNoCopy (/* CFAllocatorRef* */ IntPtr alloc, string chars, nint numChars, nint capacity, /* CFAllocatorRef* */ IntPtr externalCharactersAllocator);
+
+		public CFMutableString (string theString, nint capacity)
+		{
+			if (theString == null)
+				throw new ArgumentNullException (nameof (theString));
+			
+			handle = CFStringCreateMutableWithExternalCharactersNoCopy (IntPtr.Zero, theString, theString.Length, capacity, IntPtr.Zero);
+		}
+
+		[DllImport (Constants.CoreFoundationLibrary)]
+		static extern unsafe /* CFMutableStringRef* */ IntPtr CFStringCreateMutable (/* CFAllocatorRef* */ IntPtr alloc, nint maxLength);
+
+		public CFMutableString (nint maxLength)
+		{
+			handle = CFStringCreateMutable (IntPtr.Zero, maxLength);
```

Comment:
Maybe throw an exception if `handle` ends up being IntPtr.Zero?

Reply:
I need to test what happens if you allocate a `nil` string and append to it
It's also something that we should do (and share) on the base `CFString` type

Reply:
CFString should inherit from NativeObject, then we'd get much of this logic for free :smile:

Reply:
yeah, it's getting farther and father away from the single missing API
but I guess it's a good time to bite the bullet on this one :)

Reply:
done